tests/layer_tests/tensorflow_tests/test_tf_Gather.py

```diff
@@ -79,3 +79,81 @@ def test_gather(self, params, params_type, indices_type, ie_device, precision, i
         self._test(*self.create_gather_net(**params, params_type=params_type, indices_type=indices_type),
                    ie_device, precision, ir_version, temp_dir=temp_dir,
                    use_legacy_frontend=use_legacy_frontend)
+        
+class TestComplexGather(CommonTFLayerTest):
+    def _prepare_input(self, inputs_info):
+        assert 'real_params:0' in inputs_info
+        assert 'imag_params:0' in inputs_info
+        assert 'indices:0' in inputs_info
+        real_params_shape = inputs_info['real_params:0']
+        imag_params_shape = inputs_info['imag_params:0']
+        indices_shape = inputs_info['indices:0']
+        inputs_data = {}
+        if self.params_type == str or self.params_type == np.str_:
+            strings_dictionary = ['first', 'second sentence', ' sentence 3 three', '34ferf466 23435* ']
+            inputs_data['real_params:0'] = rng.choice(strings_dictionary, real_params_shape)
+            inputs_data['imag_params:0'] = rng.choice(strings_dictionary, imag_params_shape)
+        else:
```

Comment:
here we should test only float32 type for complex type. String type is not valid here.